crates/broker/src/order_picker.rs

```diff
@@ -810,8 +824,18 @@ where
                 tokio::select! {
                     // This channel is cancellation safe, so it's fine to use in the select!
                     Some(order) = rx.recv() => {
-                        tracing::debug!("Queued order {} to be priced", order.id());
+                        let order_id = order.id();
                         pending_orders.push(order);
+                        tracing::debug!(
+                            "Queued order {} to be priced. Currently {} queued pricing tasks: {}",
+                            order_id,
+                            pending_orders.len(),
+                            pending_orders
+                                .iter()
+                                .map(ToString::to_string)
+                                .collect::<Vec<_>>()
+                                .join(", ")
+                        );
```

Comment:
This seems like it would clutter logs a bit if a lot of order requests come in. Probably fine for now